Content.Server/GameObjects/Components/Mobs/Body/Limb.cs

```diff
@@ -0,0 +1,322 @@
+﻿using System;
+using Robust.Shared.Maths;
+using System.Collections.Generic;
+using Robust.Shared.Serialization;
+using Content.Shared.GameObjects;
+using Robust.Shared.Log;
+using Robust.Shared.Interfaces.GameObjects;
+using YamlDotNet.RepresentationModel;
+using Robust.Shared.Prototypes;
+using Content.Server.GameObjects.Components.Mobs.Body.Organs;
+using Robust.Shared.IoC;
+using System.Linq;
+using Robust.Shared.Utility;
+using Robust.Shared.Interfaces.Reflection;
+using Robust.Shared.Reflection;
+
+namespace Content.Server.GameObjects.Components.Mobs.Body
+{
+    /// <summary>
+    ///     Limb is not just like <see cref="Organ"/>, it has BONES, and holds organs (and child limbs), 
+    ///     it receive damage first, then through resistances and such it transfers the damage to organs,
+    ///     also the limb is visible, and it can be targeted
+    /// </summary>
+    [Prototype("limb")]
+    public class Limb : IPrototype, IIndexedPrototype
+    {
+#pragma warning disable CS0649
+        [Dependency]
+        protected IPrototypeManager PrototypeManager;
+        protected IReflectionManager reflectionManager;
+#pragma warning restore
+
+        public string Name;
+        public string Id;
+        public string TexturePath = "";
+        public List<Organ> Organs = new List<Organ>();
+
+        public string Parent;
+        public List<Limb> Children = new List<Limb>();
+
+        Random _seed;
+        string TargetKey;
+        AttackTargetDef AttackTarget;
+        LimbState State;
+        int MaxHealth;
+        int CurrentHealth;
+        IEntity Owner;
+        BodyTemplate BodyOwner;
+        float BloodChange = 0f;
+        string PrototypeEntity = "";
+        bool childOrganDamage; //limbs that are used for deattaching, they'll get deleted if parent is dropped. (arms, legs)
+        bool directOrganDamage;
+        List<string> _organProt = new List<string>();
+        /// <summary>
+        /// Additional targets like Mouth and Eyes.
+        /// </summary>
+        Dictionary<string, string> _addTargetKeys = new Dictionary<string, string>();
+        List<AdditionalTarget> _addTargets = new List<AdditionalTarget>();
+        List<LimbStatus> Statuses = new List<LimbStatus>();
```

Comment:
If this is what I assume it should be, you should use some sort of set datastructure like a hash set instead of a plain list here.

Also, it's unused, but you probably know that.